source/jormungandr/jormungandr/street_network/handimap.py

```diff
@@ -0,0 +1,336 @@
+            ),
+            reset_timeout=kwargs.get(
+                'circuit_breaker_reset_timeout', app.config.get('CIRCUIT_BREAKER_HANDIMAP_TIMEOUT_S', 60)
+            ),
+        )
+        self._feed_publisher = FeedPublisher(**feed_publisher) if feed_publisher else None
+
+    def status(self):
+        return {
+            'id': six.text_type(self.sn_system_id),
+            'class': self.__class__.__name__,
+            'modes': self.modes,
+            'timeout': self.timeout,
+            'circuit_breaker': {
+                'current_state': self.breaker.current_state,
+                'fail_counter': self.breaker.fail_counter,
+                'reset_timeout': self.breaker.reset_timeout,
+            },
+        }
+
+    def _get_language(self, language):
+        try:
+            return Languages[language].value
+        except KeyError:
+            self.log.error('Handimap parameters language={} not exist - force to french'.format(language))
+            return Languages.french.value
+
+    def get_language_parameter(self, request):
+        language = request.get('_handimap_language', None)
+        return self.language if not language else self._get_language(language.lower())
+
+    @classmethod
+    def _make_request_arguments_walking_details(cls, walking_speed, language):
+        walking_speed_km = mps_to_kmph(walking_speed)
+        return {
+            "costing": "walking",
+            "costing_options": {"walking": {"walking_speed": walking_speed_km}},
+            "directions_options": {"units": "kilometers", "language": language},
+        }
+
+    @classmethod
+    def _format_coord(cls, pt_object):
+        coord = get_pt_object_coord(pt_object)
+        return {"lat": coord.lat, "lon": coord.lon}
+
+    @classmethod
+    def _make_request_arguments_direct_path(cls, origin, destination, walking_speed, language):
+        walking_details = cls._make_request_arguments_walking_details(walking_speed, language)
+        params = {
+            'locations': [
+                cls._format_coord(origin),
+                cls._format_coord(destination),
+            ]
+        }
+        params.update(walking_details)
+        return params
+
+    def matrix_payload(self, origins, destinations, walking_speed, language):
+
+        walking_details = self._make_request_arguments_walking_details(walking_speed, language)
+
+        params = {
+            "sources": [self._format_coord(o) for o in origins],
+            "targets": [self._format_coord(d) for d in destinations],
+        }
+        params.update(walking_details)
+
+        return params
+
+    def post_matrix_request(self, origins, destinations, walking_speed, language):
+        post_data = self.matrix_payload(origins, destinations, walking_speed, language)
+        response = self._call_handimap(
+            'sources_to_targets',
+            post_data,
+        )
+        self.check_response(response)
+        return ujson.loads(response.text)
```

Comment:
i would recommend you to add a try-catch around `ujson.loads` in case of non utf-8 characters...